honeycombio/resource_trigger.go

```diff
@@ -261,3 +265,44 @@ func flattenTriggerRecipients(rs []honeycombio.TriggerRecipient) []map[string]in
 
 	return result
 }
+
+// Provides a stable order for Trigger recipients.
+//
+// This cannot currently be handled efficiently by a DiffSuppressFunc.
+// See: https://github.com/hashicorp/terraform-plugin-sdk/issues/477
+func matchRecipientsWithSchema(recipients []honeycombio.TriggerRecipient, declaredRecipients []interface{}) []honeycombio.TriggerRecipient {
+	result := make([]honeycombio.TriggerRecipient, len(declaredRecipients))
+
+	rMap := make(map[string]honeycombio.TriggerRecipient, len(declaredRecipients))
+	for _, recipient := range recipients {
+		rMap[recipient.ID] = recipient
+	}
+
+	for i, declaredRcpt := range declaredRecipients {
```

Comment:
I'm opening to renaming the function `matchRecipientsWithSchemaByType` to it felt .. verbose

Reply:
I'd still like a little bit more discussion here, as there's a lot of logic that's difficult to grok at first glance. How's this?

```suggestion
    // Build up result, with each readRecipient in the same position as it
    // appears in declaredRecipients, by looking at each declaredRecipient and
    // finding its matching readRecipient (via rMap).
    //
    // If the declaredRecipient has an ID, this is easy: just look it up and
    // put it in it's place. Otherwise, try to match it to a readRecipient with
    // the same type and target. If we can't find it at all, it must be new, so
    // put it at the end.
	for i, declaredRcpt := range declaredRecipients {
```

Reply:
dig it: added!